using hono requires skipLibCheck

Hi,

I've just tried adding your library to my project. However, simply running `new Hono();` breaks my typescript project due to numerous type issues with your bundled types; some of your types expect the global `Request` to be a DOM request object, but this is not true in the workers environment.

Passing `skipLibCheck: true` to my tsconfig suppresses the errors, which I notice is what the [hono repo itself does in its own tsconfig](https://github.com/honojs/hono/blob/4b2dfb824ba68c6dd03ea93ba60649a95d370e80/tsconfig.json#L10).

However, this is definitely not a good thing to do, as it disables ALL type checking for all dependencies.

Here is a demonstration. Any ideas? I've also included my tsconfig below. 

```
~/P/server main• 1s ❱ tsc --noEmit --project .
~/P/server main• ❱ echo "Note: no type errors in my project prior to adding hono"
Note: no type errors in my project prior to adding hono
~/P/server main• ❱ npm install hono

added 1 package, and audited 915 packages in 883ms

152 packages are looking for funding
  run `npm fund` for details

1 moderate severity vulnerability

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
~/P/server main• 1s ❱ echo -e "import { Hono } from 'hono';\n const app = new Hono();" >> src/worker.ts
~/P/server main• ❱ tsc --noEmit --project .
../node_modules/hono/dist/types/client/types.d.ts:22:18 - error TS2430: Interface 'ClientResponse<T>' incorrectly extends interface 'Response'.
  The types returned by 'json()' are incompatible between these types.
    Type 'Promise<T>' is not assignable to type 'Promise<T>'. Two different types with this name exist, but they are unrelated.
      Type 'T' is not assignable to type 'T'. Two different types with this name exist, but they are unrelated.
        'T' could be instantiated with an arbitrary type which could be unrelated to 'T'.

22 export interface ClientResponse<T> extends Response {
                    ~~~~~~~~~~~~~~

  ../node_modules/hono/dist/types/client/types.d.ts:22:33
    22 export interface ClientResponse<T> extends Response {
                                       ~
    This type parameter might need an `extends T` constraint.

../node_modules/hono/dist/types/request.d.ts:51:21 - error TS2304: Cannot find name 'RequestRedirect'.

51     get redirect(): RequestRedirect;
                       ~~~~~~~~~~~~~~~

../node_modules/hono/dist/types/request.d.ts:54:18 - error TS2304: Cannot find name 'RequestCache'.

54     get cache(): RequestCache;
                    ~~~~~~~~~~~~

../node_modules/hono/dist/types/request.d.ts:55:24 - error TS2304: Cannot find name 'RequestCredentials'.

55     get credentials(): RequestCredentials;
                          ~~~~~~~~~~~~~~~~~~

../node_modules/hono/dist/types/request.d.ts:58:17 - error TS2304: Cannot find name 'RequestMode'.

58     get mode(): RequestMode;
                   ~~~~~~~~~~~

../node_modules/hono/dist/types/request.d.ts:60:26 - error TS2304: Cannot find name 'ReferrerPolicy'.

60     get refererPolicy(): ReferrerPolicy;
                            ~~~~~~~~~~~~~~


Found 6 errors in 2 files.

Errors  Files
     1  ../node_modules/hono/dist/types/client/types.d.ts:22
     5  ../node_modules/hono/dist/types/request.d.ts:51
~/P/server main• | 2 ❱
```

```tsconfig
...
  "compilerOptions": {
    "lib": ["es2022"],
    "types": ["@cloudflare/workers-types"]
  },
...
  ```

--- a/deno_dist/adapter/deno/serve-static.ts
+++ b/deno_dist/adapter/deno/serve-static.ts
@@ -3,6 +3,10 @@ import type { Next } from '../../types.ts'
 import { getFilePath } from '../../utils/filepath.ts'
 import { getMimeType } from '../../utils/mime.ts'
 
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+const { readFile } = Deno
+
 export type ServeStaticOptions = {
   root?: string
   path?: string
@@ -32,7 +36,7 @@ export const serveStatic = (options: ServeStaticOptions = { root: '' }) => {
     let content
 
     try {
-      content = await Deno.readFile(path)
+      content = await readFile(path)
     } catch (e) {
       console.warn(`${e}`)
     }

--- a/package.json
+++ b/package.json
@@ -350,7 +350,7 @@
   ],
   "devDependencies": {
     "@cloudflare/workers-types": "^4.20221111.1",
-    "@hono/node-server": "^0.2.3",
+    "@hono/node-server": "^1.0.2",
     "@types/crypto-js": "^4.1.1",
     "@types/glob": "^8.0.0",
     "@types/jest": "^29.4.0",

--- a/src/adapter/aws-lambda/handler.ts
+++ b/src/adapter/aws-lambda/handler.ts
@@ -6,7 +6,7 @@ import { encodeBase64 } from '../../utils/encode'
 
 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
 // @ts-ignore
-globalThis.crypto = crypto
+globalThis.crypto ??= crypto
 
 // When calling Lambda directly through function urls
 interface APIGatewayProxyEventV2 {

--- a/src/adapter/deno/serve-static.ts
+++ b/src/adapter/deno/serve-static.ts
@@ -3,6 +3,10 @@ import type { Next } from '../../types'
 import { getFilePath } from '../../utils/filepath'
 import { getMimeType } from '../../utils/mime'
 
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+const { readFile } = Deno
+
 export type ServeStaticOptions = {
   root?: string
   path?: string
@@ -32,7 +36,7 @@ export const serveStatic = (options: ServeStaticOptions = { root: '' }) => {
     let content
 
     try {
-      content = await Deno.readFile(path)
+      content = await readFile(path)
     } catch (e) {
       console.warn(`${e}`)
     }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -7,12 +7,9 @@
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
     "strict": true,
-    "skipLibCheck": true,
+    "skipLibCheck": false,
     "noUnusedLocals": false,
     "noUnusedParameters": false,
-    "lib": [
-      "WebWorker"
-    ],
     "types": [
       "jest",
       "node"

